opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -12,141 +12,123 @@
-        thread = Thread(
-            target=Context.with_current_context(work),
-            args=('bar',),
-        )
-        thread.start()
-        thread.join()
-
-        print('Main thread:', Context)
-
-Here goes another example using thread pool::
-
-    import time
-    import threading
-
-    from multiprocessing.dummy import Pool as ThreadPool
-    from opentelemetry.context import Context
-
-    _console_lock = threading.Lock()
-
-    def println(msg):
-        with _console_lock:
-            print(msg)
-
-    def work(name):
-        println('Entering worker[{}]: {}'.format(name, Context))
-        Context.operation_id = name
-        time.sleep(0.01)
-        println('Exiting worker[{}]: {}'.format(name, Context))
-
-    if __name__ == "__main__":
-        println('Main thread: {}'.format(Context))
-        Context.operation_id = 'main'
-        pool = ThreadPool(2)  # create a thread pool with 2 threads
-        pool.map(Context.with_current_context(work), [
-            'bear',
-            'cat',
-            'dog',
-            'horse',
-            'rabbit',
-        ])
-        pool.close()
-        pool.join()
-        println('Main thread: {}'.format(Context))
-
-Here goes a simple demo of how async could work in Python 3.7+::
-
-    import asyncio
-
-    from opentelemetry.context import Context
-
-    class Span(object):
-        def __init__(self, name):
-            self.name = name
-            self.parent = Context.current_span
-
-        def __repr__(self):
-            return ('{}(name={}, parent={})'
-                    .format(
-                        type(self).__name__,
-                        self.name,
-                        self.parent,
-                    ))
-
-        async def __aenter__(self):
-            Context.current_span = self
-
-        async def __aexit__(self, exc_type, exc, tb):
-            Context.current_span = self.parent
-
-    async def main():
-        print(Context)
-        async with Span('foo'):
-            print(Context)
-            await asyncio.sleep(0.1)
-            async with Span('bar'):
-                print(Context)
-                await asyncio.sleep(0.1)
-            print(Context)
-            await asyncio.sleep(0.1)
-        print(Context)
-
-    if __name__ == '__main__':
-        asyncio.run(main())
-"""
-
-from .base_context import BaseRuntimeContext
-
-__all__ = ["Context"]
-
-try:
-    from .async_context import AsyncRuntimeContext
-
-    Context = AsyncRuntimeContext()  # type: BaseRuntimeContext
-except ImportError:
-    from .thread_local_context import ThreadLocalRuntimeContext
-
-    Context = ThreadLocalRuntimeContext()
+import logging
+import typing
+from os import environ
+
+from pkg_resources import iter_entry_points
+
+from opentelemetry.context.context import Context
+
+logger = logging.getLogger(__name__)
+_CONTEXT = None  # type: typing.Optional[Context]
+
+
+def _copy_context(context: typing.Optional[Context]) -> Context:
```

Comment:
I'd prefer writing these methods in the form
```py
if not context:
  context = get_current()
return some_action(context)
```

or even `return some_action(context or get_current())`.